From 16ba54387d106d24aaacec9e23c810701447236f Mon Sep 17 00:00:00 2001
From: Elmar Stein <elmar.stein@sennheiser.com>
Date: Tue, 30 Jul 2019 17:48:48 +0200
Subject: [PATCH] Updated HCI Parser for parsing new LE Meta Events

---
 tools/parser/hci.c | 25 +++++++++++++++++++++++--
 1 file changed, 23 insertions(+), 2 deletions(-)

diff --git a/tools/parser/hci.c b/tools/parser/hci.c
index 8c7bd25..643dc8b 100644
--- a/tools/parser/hci.c
+++ b/tools/parser/hci.c
@@ -127,7 +127,7 @@ static char *event_str[EVENT_NUM + 1] = {
 	"AMP Status Change",
 };
 
-#define LE_EV_NUM 5
+#define LE_EV_NUM 24
 static char *ev_le_meta_str[LE_EV_NUM + 1] = {
 	"Unknown",
 	"LE Connection Complete",
@@ -135,6 +135,24 @@ static char *ev_le_meta_str[LE_EV_NUM + 1] = {
 	"LE Connection Update Complete",
 	"LE Read Remote Used Features Complete",
 	"LE Long Term Key Request",
+	"LE Remote Connection Parameter Request",
+	"LE Data Length Change Event",
+	"LE Read Local P-256 Public Key Complete event",
+	"LE Generate DHKey Complete event",
+	"LE Enhanced Connection Complete event",
+	"LE Directed Advertising Report event",
+	"LE PHY Update Complete event",
+	"LE Extended Advertising Report event",
+	"LE Periodic Advertising Sync Established event",
+	"LE Periodic Advertising Report event",
+	"LE Periodic Advertising Sync Lost event",
+	"LE Scan Timeout event",
+	"LE Advertising Set Terminated event",
+	"LE Scan Request Received event",
+	"LE Channel Selection Algorithm event",
+	"LE Connectionless IQ Report event",
+	"LE Connection IQ Report event",
+	"LE CTE Request Failed event",
 };
 
 #define CMD_LINKCTL_NUM 60
@@ -3667,7 +3685,10 @@ static inline void le_meta_ev_dump(int level, struct frame *frm)
 	frm->len -= EVT_LE_META_EVENT_SIZE;
 
 	p_indent(level, frm);
-	printf("%s\n", ev_le_meta_str[subevent]);
+	if (subevent <= LE_EV_NUM)
+		printf("%s\n", ev_le_meta_str[subevent]);
+	else
+		printf("Unknown Meta LE subevent\n");
 
 	switch (mevt->subevent) {
 	case EVT_LE_CONN_COMPLETE:
-- 
2.7.4

