From a6ce8f21b980cbad8f2b783f80462c2371fe1485 Mon Sep 17 00:00:00 2001
From: Simon Mikuda <simon.mikuda@streamunlimited.com>
Date: Wed, 18 Sep 2019 12:41:53 +0200
Subject: [PATCH] device: Start discovery when there are no records for
 discovered services

Signed-off-by: Simon Mikuda <simon.mikuda@streamunlimited.com>
---
 src/device.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/device.c b/src/device.c
index 49d45d9..2b47f19 100644
--- a/src/device.c
+++ b/src/device.c
@@ -5778,6 +5778,15 @@ unsigned int device_wait_for_svc_complete(struct btd_device *dev,
 
 	dev->svc_callbacks = g_slist_prepend(dev->svc_callbacks, cb);
 
+	if (!dev->tmp_records && state->svc_resolved) {
+		if (dev->discov_timer > 0)
+			g_source_remove(dev->discov_timer);
+		else
+			DBG("No records found for resolved services - start discovery");
+		dev->discov_timer = g_idle_add(start_discovery, dev);
+		return cb->id;
+	}
+
 	if (state->svc_resolved || !main_opts.reverse_sdp)
 		cb->idle_id = g_idle_add(svc_idle_cb, cb);
 	else if (dev->discov_timer > 0) {
-- 
2.7.4

