From 241ab25fea0c45306307442e903cc0bb2b17cd4d Mon Sep 17 00:00:00 2001
From: Marek Belisko <marek.belisko@gmail.com>
Date: Thu, 8 Oct 2015 08:57:49 +0200
Subject: [PATCH] board-spl: Add support for DDR3

Signed-off-by: Marek Belisko <marek.belisko@streamunlimited.com>
---
 board/streamunlimited/stream800/board-spl.c | 125 ++++++++--------------------
 1 file changed, 33 insertions(+), 92 deletions(-)

diff --git a/board/streamunlimited/stream800/board-spl.c b/board/streamunlimited/stream800/board-spl.c
index 015bb51..2317e18 100644
--- a/board/streamunlimited/stream800/board-spl.c
+++ b/board/streamunlimited/stream800/board-spl.c
@@ -43,61 +43,40 @@
 
 DECLARE_GLOBAL_DATA_PTR;
 
-/* Definitions for DDR2 */
-
-static const struct ddr_data ddr2_data = {
-	.datardsratio0 = ((MT47H128M16RT25E_RD_DQS<<30) |
-			  (MT47H128M16RT25E_RD_DQS<<20) |
-			  (MT47H128M16RT25E_RD_DQS<<10) |
-			  (MT47H128M16RT25E_RD_DQS<<0)),
-	.datawdsratio0 = ((MT47H128M16RT25E_WR_DQS<<30) |
-			  (MT47H128M16RT25E_WR_DQS<<20) |
-			  (MT47H128M16RT25E_WR_DQS<<10) |
-			  (MT47H128M16RT25E_WR_DQS<<0)),
-	.datawiratio0 = ((MT47H128M16RT25E_PHY_WRLVL<<30) |
-			 (MT47H128M16RT25E_PHY_WRLVL<<20) |
-			 (MT47H128M16RT25E_PHY_WRLVL<<10) |
-			 (MT47H128M16RT25E_PHY_WRLVL<<0)),
-	.datagiratio0 = ((MT47H128M16RT25E_PHY_GATELVL<<30) |
-			 (MT47H128M16RT25E_PHY_GATELVL<<20) |
-			 (MT47H128M16RT25E_PHY_GATELVL<<10) |
-			 (MT47H128M16RT25E_PHY_GATELVL<<0)),
-	.datafwsratio0 = ((MT47H128M16RT25E_PHY_FIFO_WE<<30) |
-			  (MT47H128M16RT25E_PHY_FIFO_WE<<20) |
-			  (MT47H128M16RT25E_PHY_FIFO_WE<<10) |
-			  (MT47H128M16RT25E_PHY_FIFO_WE<<0)),
-	.datawrsratio0 = ((MT47H128M16RT25E_PHY_WR_DATA<<30) |
-			  (MT47H128M16RT25E_PHY_WR_DATA<<20) |
-			  (MT47H128M16RT25E_PHY_WR_DATA<<10) |
-			  (MT47H128M16RT25E_PHY_WR_DATA<<0)),
+static const struct ddr_data ddr3_ase_data = {
+	.datardsratio0 = MT41K256M16HA125E_RD_DQS,
+	.datawdsratio0 = MT41K256M16HA125E_WR_DQS,
+	.datafwsratio0 = MT41K256M16HA125E_PHY_FIFO_WE,
+	.datawrsratio0 = MT41K256M16HA125E_PHY_WR_DATA,
 };
 
-static const struct cmd_control ddr2_cmd_ctrl_data = {
-	.cmd0csratio = MT47H128M16RT25E_RATIO,
-	.cmd0iclkout = MT47H128M16RT25E_INVERT_CLKOUT,
+static const struct cmd_control ddr3_ase_cmd_ctrl_data = {
+	.cmd0csratio = MT41K256M16HA125E_RATIO,
+	.cmd0iclkout = MT41K256M16HA125E_INVERT_CLKOUT,
 
-	.cmd1csratio = MT47H128M16RT25E_RATIO,
-	.cmd1iclkout = MT47H128M16RT25E_INVERT_CLKOUT,
+	.cmd1csratio = MT41K256M16HA125E_RATIO,
+	.cmd1iclkout = MT41K256M16HA125E_INVERT_CLKOUT,
 
-	.cmd2csratio = MT47H128M16RT25E_RATIO,
-	.cmd2iclkout = MT47H128M16RT25E_INVERT_CLKOUT,
+	.cmd2csratio = MT41K256M16HA125E_RATIO,
+	.cmd2iclkout = MT41K256M16HA125E_INVERT_CLKOUT,
 };
 
-static const struct emif_regs ddr2_emif_reg_data = {
-	.sdram_config		= MT47H128M16RT25E_EMIF_SDCFG,
-	.ref_ctrl		= MT47H128M16RT25E_EMIF_SDREF,
-	.sdram_tim1		= MT47H128M16RT25E_EMIF_TIM1,
-	.sdram_tim2		= MT47H128M16RT25E_EMIF_TIM2,
-	.sdram_tim3		= MT47H128M16RT25E_EMIF_TIM3,
-	.emif_ddr_phy_ctlr_1	= MT47H128M16RT25E_EMIF_READ_LATENCY,
+static struct emif_regs ddr3_ase_emif_reg_data = {
+	.sdram_config = MT41K256M16HA125E_EMIF_SDCFG,
+	.ref_ctrl = MT41K256M16HA125E_EMIF_SDREF,
+	.sdram_tim1 = MT41K256M16HA125E_EMIF_TIM1,
+	.sdram_tim2 = MT41K256M16HA125E_EMIF_TIM2,
+	.sdram_tim3 = MT41K256M16HA125E_EMIF_TIM3,
+	.zq_config = MT41K256M16HA125E_ZQ_CFG,
+	.emif_ddr_phy_ctlr_1 = MT41K256M16HA125E_EMIF_READ_LATENCY,
 };
 
-static const struct ctrl_ioregs ioregs_ddr2 = {
-	.cm0ioctl		= MT47H128M16RT25E_IOCTRL_VALUE,
-	.cm1ioctl		= MT47H128M16RT25E_IOCTRL_VALUE,
-	.cm2ioctl		= MT47H128M16RT25E_IOCTRL_VALUE,
-	.dt0ioctl		= MT47H128M16RT25E_IOCTRL_VALUE,
-	.dt1ioctl		= MT47H128M16RT25E_IOCTRL_VALUE,
+const struct ctrl_ioregs ioregs_ase = {
+	.cm0ioctl		= MT41K256M16HA125E_IOCTRL_VALUE,
+	.cm1ioctl		= MT41K256M16HA125E_IOCTRL_VALUE,
+	.cm2ioctl		= MT41K256M16HA125E_IOCTRL_VALUE,
+	.dt0ioctl		= MT41K256M16HA125E_IOCTRL_VALUE,
+	.dt1ioctl		= MT41K256M16HA125E_IOCTRL_VALUE,
 };
 
 /* Definitions for DDR3 */
@@ -152,17 +131,6 @@ static const struct cmd_control ddr3_cmd_ctrl_data = {
 	.cmd2iclkout	= MT41K128M16JT125E_303_INVERT_CLKOUT,
 };
 
-static struct emif_regs ddr3_emif_reg_data = {
-	.sdram_config	     = MT41K128M16JT125E_303_EMIF_SDCFG,
-	.ref_ctrl	     = MT41K128M16JT125E_303_EMIF_SDREF,
-	.sdram_tim1	     = MT41K128M16JT125E_303_EMIF_TIM1,
-	.sdram_tim2	     = MT41K128M16JT125E_303_EMIF_TIM2,
-	.sdram_tim3	     = MT41K128M16JT125E_303_EMIF_TIM3,
-	.zq_config	     = MT41K128M16JT125E_303_ZQ_CFG,
-	.emif_ddr_phy_ctlr_1 = (MT41K128M16JT125E_303_EMIF_READ_LATENCY |
-				PHY_EN_DYN_PWRDN),
-};
-
 static const struct ctrl_ioregs ioregs_ddr3 = {
 	.cm0ioctl	= MT41K128M16JT125E_303_IOCTRL_VALUE,
 	.cm1ioctl	= MT41K128M16JT125E_303_IOCTRL_VALUE,
@@ -201,43 +169,16 @@ void set_uart_mux_conf(void)
 
 void sdram_init(void)
 {
-	struct Stream800Board board;
-	init_adc();
-	board_detect(&board);
-	board_printf_info(&board);
-
-	switch (board.moduleVersion) {
-	case MV_L1_DDR3_HE:
-	case MV_L0_DDR3:
-		config_ddr(303, &ioregs_ddr3, &ddr3_data,
-			   &ddr3_cmd_ctrl_data, &ddr3_emif_reg_data, 0);
-		break;
-	case MV_unknown:
-	case MV_L0:
-	case MV_L1:
-	case MV_L2:
-	default:
-		config_ddr(266, &ioregs_ddr2, &ddr2_data,
-			   &ddr2_cmd_ctrl_data, &ddr2_emif_reg_data, 0);
-		break;
-	}
-
-	/* As we've just done board detection this is a useful spot to do the test
-	 * for a factory carrier board.  If so, and we booted from UART, then load
-	 * u-boot.img from mmc1. */
-	if (spl_boot_device() == BOOT_DEVICE_UART &&
-	    board.carrierBoardType == CBT_SueFactoryTestBoard) {
-		printf("Running on factory test carrier and booted via UART, loading u-boot.img from mmc1\n");
-		/* mmc1 must be enabled before it initialised. */
-		enable_mmc1_pin_mux();
-		/* Force mmc1 as boot device and set mode to FAT. */
-		gd->arch.omap_boot_params.omap_bootdevice = BOOT_DEVICE_MMC2;
-		gd->arch.omap_boot_params.omap_bootmode = MMCSD_MODE_FAT;
-	}
+	config_ddr(400, &ioregs_ase,
+			   &ddr3_ase_data,
+			   &ddr3_ase_cmd_ctrl_data,
+			   &ddr3_ase_emif_reg_data, 0);
+
 }
 
 #define OSC	(V_OSCK/1000000)
-const struct dpll_params dpll_ddr = {266, OSC-1, 1, -1, -1, -1, -1};
+const struct dpll_params dpll_ddr = {
+		400, OSC-1, 1, -1, -1, -1, -1};
 
 const struct dpll_params *get_dpll_ddr_params(void)
 {
-- 
2.7.4

