From 0dfc6bf4a247fae0d9d9d0ce835d133c8f79988c Mon Sep 17 00:00:00 2001
From: Martin Flaska <martin.flaska@streamunlimited.com>
Date: Tue, 29 Jan 2013 07:57:17 +0100
Subject: [PATCH] stream800: enable I/D caches

Signed-off-by: Martin Flaska <martin.flaska@streamunlimited.com>
---
 board/streamunlimited/stream800/board.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/board/streamunlimited/stream800/board.c b/board/streamunlimited/stream800/board.c
index 6f5f5db..bb9afa3 100644
--- a/board/streamunlimited/stream800/board.c
+++ b/board/streamunlimited/stream800/board.c
@@ -71,6 +71,12 @@ static struct Stream800Board __attribute__((section (".data"))) board = {moduleV
 
 static void init_si5351x(void);
 
+void enable_caches(void)
+{
+	puts("Enabling caches ...\n");
+	icache_enable();
+	dcache_enable();
+}
 
 Stream800ModuleVersion get_module_version(void)
 {
-- 
2.7.4

