From 2c7d6c276d50d574425cef27eba1617496940bb9 Mon Sep 17 00:00:00 2001
From: Florin Hillebrand <florin.hillebrand@streamunlimited.com>
Date: Tue, 12 Nov 2013 16:10:03 +0100
Subject: [PATCH] stream800: L2 - Stream800/2E8-4 module config.

Signed-off-by: Florin Hillebrand <florin.hillebrand@streamunlimited.com>
---
 board/streamunlimited/stream800/board.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/board/streamunlimited/stream800/board.c b/board/streamunlimited/stream800/board.c
index 941e7ab..3cfaec7 100755
--- a/board/streamunlimited/stream800/board.c
+++ b/board/streamunlimited/stream800/board.c
@@ -201,6 +201,8 @@ static struct SupportedModules {
 	{MV_L2,		0xCCC,  0xD54,  0x5dd,  0x666},
 	// L2 - Stream800/2E7-4 (no wifi/bt module, 4G flash)
 	{MV_L2,		0xCCC,  0xD54,  0x6ee,  0x777},
+	// L2 - Stream800/2E8-4 (no wifi/bt module, 800Mhz, 4G flash)
+	{MV_L2,		0xCCC,  0xD54,  0x777,  0x7ff},
 	// TODO: add another stream800 module revisions here
 };
 
-- 
2.7.4

