From 2b15ba85f87e4d79add3349926b80f5e3f6b8dac Mon Sep 17 00:00:00 2001
From: Simon Mikuda <simon.mikuda@streamunlimited.com>
Date: Fri, 6 Nov 2020 10:14:32 +0100
Subject: [PATCH] autogen.sh

---
 autogen.sh | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/autogen.sh b/autogen.sh
index 346605d..a934a11 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -5,7 +5,6 @@ srcdir=`dirname $0`
 test -z "$srcdir" && srcdir=.
 
 PKG_NAME="totem-pl-parser"
-REQUIRED_M4MACROS=introspection.m4
 
 (test -f $srcdir/configure.ac) || {
     echo -n "**Error**: Directory "\`$srcdir\'" does not look like the"
@@ -13,9 +12,5 @@ REQUIRED_M4MACROS=introspection.m4
     exit 1
 }
 
-which gnome-autogen.sh || {
-	echo "You need to install gnome-common from the GNOME SVN"
-	exit 1
-}
+autoreconf -i -f
 
-REQUIRED_PKG_CONFIG_VERSION=0.17.1 REQUIRED_AUTOMAKE_VERSION=1.9 USE_GNOME2_MACROS=1 . gnome-autogen.sh --enable-gtk-doc "$@"
