From 3083ab305f863c40dfe2570f5c94fe1771449b04 Mon Sep 17 00:00:00 2001
From: Simon Mikuda <simon.mikuda@streamunlimited.com>
Date: Fri, 13 Mar 2020 09:28:18 +0100
Subject: [PATCH] Add switch to internal plugins list

Signed-off-by: Simon Mikuda <simon.mikuda@streamunlimited.com>
---
 configure    | 2 +-
 configure.ac | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure b/configure
index 61d874f..0749ae2 100755
--- a/configure
+++ b/configure
@@ -13014,7 +13014,7 @@ fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $gcc_have_atomics" >&5
 $as_echo "$gcc_have_atomics" >&6; }
 
-PCM_PLUGIN_LIST="copy linear route mulaw alaw adpcm rate plug multi shm file null empty share meter hooks lfloat ladspa dmix dshare dsnoop asym iec958 softvol extplug ioplug mmap_emul"
+PCM_PLUGIN_LIST="copy linear route mulaw alaw adpcm rate plug multi shm file null empty share meter hooks lfloat ladspa dmix dshare dsnoop asym iec958 softvol extplug ioplug mmap_emul switch"
 
 build_pcm_plugin="no"
 for t in $PCM_PLUGIN_LIST; do
diff --git a/configure.ac b/configure.ac
index cce195a..43011d1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -513,7 +513,7 @@ if test -z "$gcc_have_atomics"; then
 fi
 AC_MSG_RESULT($gcc_have_atomics)
 
-PCM_PLUGIN_LIST="copy linear route mulaw alaw adpcm rate plug multi shm file null empty share meter hooks lfloat ladspa dmix dshare dsnoop asym iec958 softvol extplug ioplug mmap_emul"
+PCM_PLUGIN_LIST="copy linear route mulaw alaw adpcm rate plug multi shm file null empty share meter hooks lfloat ladspa dmix dshare dsnoop asym iec958 softvol extplug ioplug mmap_emul switch"
 
 build_pcm_plugin="no"
 for t in $PCM_PLUGIN_LIST; do
-- 
2.7.4

